Unify message context, use `MooseBase` with `MooseApp`, remove deprecated registration

loganharbour opened the pull request:
Closes #30203
Closes #30778

- Combines parameter and error context `MooseBase`, which enables `MooseApp` to get these things
- Adds more error context to `InputParameters`, so that errors like a parameter not found will have the context of the object if available
- Removes deprecated registration in MooseApp`
- Attempts to force const parameters in `MooseApp`
- Enables calling things like `parameters().paramError(...)` without an object, which lets you give parameter errors before the object is constructed (with object context, if possible)
- Incorporates neml2 changes in https://github.com/idaholab/moose/pull/31311

### App patches

- https://github.inl.gov/ncrc/griffin/pull/2680
- https://github.inl.gov/ncrc/bison/pull/6504
- https://github.com/idaholab/TMAP8/pull/287

loganharbour committed:
Rename errorPrefix, remove unnecessary context

refs #30203

loganharbour committed:
Modify error given error message change

refs #30203

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2782097/), step Docs: sync website on 2161452 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/30204/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2782113/), step Generate coverage on 2161452 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>48c61a</th>
      <th colspan=4>#30204 216145</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.28%</td>
      <td>85.27%</td>
      <td>-0.02%</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>109484</td>
      <td>109465</td>
      <td>-19</td>
      <td>12</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>18894</td>
      <td>18914</td>
      <td>+20</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/30204/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/30204/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/stochastic_tools)
- [`subchannel`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/subchannel)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/30204/coverage/xfem)

</details>



This comment will be updated on new commits.

loganharbour committed:
Re-add deprecated errorPrefix message

cticenhour commented on framework/include/base/MooseBase.h:
```diff
@@ -63,18 +63,23 @@ class MooseBase
 
   /**
-   * @returns A prefix to be used in errors that contains the input
+   * @returns A prefix to be used in messages that contains the input
```
```suggestion
   * @returns A prefix to be used in messages that contain the input
```

cticenhour reviewed (changes requested):
One minor adjustment to Doxygen doco, but code looks good.

loganharbour committed:
Link InputParameters to an object when possible and improve errors

refs #30203

GiudGiud commented:
Test failure is related
https://civet.inl.gov/job/2784293/

loganharbour commented:
Griffin patch is in devel, pending a master merge (currently broken due to unrelated failures)

loganharbour committed:
Store hit context separately in mooseError

This will let us capture it in MooseServer in the future

refs #30203

loganharbour committed:
Unify MooseBase for MooseApp

refs #30778

loganharbour committed:
Use paramError from the parameters

refs #30778

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2955668/), step Clang format on ba0e978 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/30204/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/30204/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 8935b196c6cdf23ff96d5f8694945c003a12e2a1`

loganharbour committed:
Correct getParam()

refs #30778

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2955774/), step Clang format on 2a6245c wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/30204/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/30204/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 8935b196c6cdf23ff96d5f8694945c003a12e2a1`

loganharbour committed:
Remove parenthesis around param error

refs #30778

loganharbour committed:
Rework how most errors are captured and reported

This is likely a breaking change for many apps. Greater
context has been added to errors, making an attempt to associate
input context to errors as much as possible

refs #30778

loganharbour committed:
Update patch

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2967194/), step Versioner verify on 49c434c wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 59 influential files, 5 changed, 1 added, 0 removed

| package   | status   | file                                    |
|-----------|----------|-----------------------------------------|
| wasp      | CHANGE   | conda/wasp/meta.yaml                    |
| wasp      | CHANGE   | conda/wasp/pyhit.patch                  |
| wasp      | NEW      | framework/contrib/hit                   |
| wasp      | CHANGE   | python/pyhit/pyhit.py                   |
| moose-dev | CHANGE   | conda/moose-dev/conda_build_config.yaml |
| moose-dev | CHANGE   | conda/moose-dev/meta.yaml               |

### Versioner versions

Found 9 packages, 2 changed, 0 failed

| package   | status   | hash    | to hash   | version            | to version         |
|-----------|----------|---------|-----------|--------------------|--------------------|
| wasp      | CHANGE   | 3e37a23 | 8d19281   | 2025.05.13 build 1 | 2025.05.19 build 1 |
| moose-dev | CHANGE   | 6416dfc | e0e6556   | 2025.06.13         | 2025.06.19         |

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2967297/), step Versioner verify on 40380ad wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 59 influential files, 5 changed, 1 added, 0 removed

| package   | status   | file                                    |
|-----------|----------|-----------------------------------------|
| wasp      | CHANGE   | conda/wasp/meta.yaml                    |
| wasp      | CHANGE   | conda/wasp/pyhit.patch                  |
| wasp      | NEW      | framework/contrib/hit                   |
| wasp      | CHANGE   | python/pyhit/pyhit.py                   |
| moose-dev | CHANGE   | conda/moose-dev/conda_build_config.yaml |
| moose-dev | CHANGE   | conda/moose-dev/meta.yaml               |

### Versioner versions

Found 9 packages, 2 changed, 0 failed

| package   | status   | hash    | to hash   | version            | to version         |
|-----------|----------|---------|-----------|--------------------|--------------------|
| wasp      | CHANGE   | 3e37a23 | 8d19281   | 2025.05.13 build 1 | 2025.05.19 build 1 |
| moose-dev | CHANGE   | 6416dfc | e0e6556   | 2025.06.13         | 2025.06.19         |

loganharbour committed:
Address compiler errors

loganharbour committed:
Remove scrape that is no longer required with WASP update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2969671/), step Versioner verify on 4603cf9 wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 59 influential files, 7 changed, 1 added, 0 removed

| package   | status   | file                                    |
|-----------|----------|-----------------------------------------|
| wasp      | CHANGE   | conda/wasp/meta.yaml                    |
| wasp      | CHANGE   | conda/wasp/pyhit.patch                  |
| wasp      | NEW      | framework/contrib/hit                   |
| wasp      | CHANGE   | framework/contrib/wasp                  |
| wasp      | CHANGE   | python/pyhit/pyhit.py                   |
| moose-dev | CHANGE   | conda/moose-dev/conda_build_config.yaml |
| moose-dev | CHANGE   | conda/moose-dev/meta.yaml               |
| moose-dev | CHANGE   | framework/contrib/neml2                 |

### Versioner versions

Found 9 packages, 2 changed, 0 failed

| package   | status   | hash    | to hash   | version            | to version         |
|-----------|----------|---------|-----------|--------------------|--------------------|
| wasp      | CHANGE   | 3e37a23 | 78ac097   | 2025.05.13 build 1 | 2025.06.20 build 0 |
| moose-dev | CHANGE   | 6416dfc | 5ef0acf   | 2025.06.13         | 2025.06.20         |

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2969780/), step Versioner verify on 385c9d7 wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 59 influential files, 7 changed, 1 added, 0 removed

| package   | status   | file                                    |
|-----------|----------|-----------------------------------------|
| wasp      | CHANGE   | conda/wasp/meta.yaml                    |
| wasp      | CHANGE   | conda/wasp/pyhit.patch                  |
| wasp      | NEW      | framework/contrib/hit                   |
| wasp      | CHANGE   | framework/contrib/wasp                  |
| wasp      | CHANGE   | python/pyhit/pyhit.py                   |
| moose-dev | CHANGE   | conda/moose-dev/conda_build_config.yaml |
| moose-dev | CHANGE   | conda/moose-dev/meta.yaml               |
| moose-dev | CHANGE   | framework/contrib/neml2                 |

### Versioner versions

Found 9 packages, 2 changed, 0 failed

| package   | status   | hash    | to hash   | version            | to version         |
|-----------|----------|---------|-----------|--------------------|--------------------|
| wasp      | CHANGE   | 3e37a23 | 33ce26b   | 2025.05.13 build 1 | 2025.06.20 build 0 |
| moose-dev | CHANGE   | 6416dfc | 4f34b54   | 2025.06.13         | 2025.06.20         |

loganharbour committed:
Address compiler and test errors

refs #30778

loganharbour committed:
Make app input parameter storage non-name based

refs #30778

loganharbour committed:
Fix and improve tests for array variable naming

refs #30778

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2971277/), step Versioner verify on fdc3479 wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 59 influential files, 7 changed, 1 added, 0 removed

| package   | status   | file                                    |
|-----------|----------|-----------------------------------------|
| wasp      | CHANGE   | conda/wasp/meta.yaml                    |
| wasp      | CHANGE   | conda/wasp/pyhit.patch                  |
| wasp      | NEW      | framework/contrib/hit                   |
| wasp      | CHANGE   | framework/contrib/wasp                  |
| wasp      | CHANGE   | python/pyhit/pyhit.py                   |
| moose-dev | CHANGE   | conda/moose-dev/conda_build_config.yaml |
| moose-dev | CHANGE   | conda/moose-dev/meta.yaml               |
| moose-dev | CHANGE   | framework/contrib/neml2                 |

### Versioner versions

Found 9 packages, 2 changed, 0 failed

| package   | status   | hash    | to hash   | version            | to version         |
|-----------|----------|---------|-----------|--------------------|--------------------|
| wasp      | CHANGE   | 3e37a23 | 33ce26b   | 2025.05.13 build 1 | 2025.06.20 build 0 |
| moose-dev | CHANGE   | 6416dfc | 4f34b54   | 2025.06.13         | 2025.06.20         |

loganharbour committed:
Add testing for range checked params

loganharbour committed:
Add testing for AppFactory

loganharbour committed:
Remove unused moose server piece

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2975787/), step Versioner verify on 4ee4d0a wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 59 influential files, 7 changed, 1 added, 0 removed

| package   | status   | file                                    |
|-----------|----------|-----------------------------------------|
| wasp      | CHANGE   | conda/wasp/meta.yaml                    |
| wasp      | CHANGE   | conda/wasp/pyhit.patch                  |
| wasp      | NEW      | framework/contrib/hit                   |
| wasp      | CHANGE   | framework/contrib/wasp                  |
| wasp      | CHANGE   | python/pyhit/pyhit.py                   |
| moose-dev | CHANGE   | conda/moose-dev/conda_build_config.yaml |
| moose-dev | CHANGE   | conda/moose-dev/meta.yaml               |
| moose-dev | CHANGE   | framework/contrib/neml2                 |

### Versioner versions

Found 9 packages, 2 changed, 0 failed

| package   | status   | hash    | to hash   | version            | to version         |
|-----------|----------|---------|-----------|--------------------|--------------------|
| wasp      | CHANGE   | 3e37a23 | 33ce26b   | 2025.05.13 build 1 | 2025.06.20 build 0 |
| moose-dev | CHANGE   | 6416dfc | 4f34b54   | 2025.06.13         | 2025.06.20         |

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2977570/), step Versioner verify on e8ab7fc wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 59 influential files, 7 changed, 1 added, 0 removed

| package   | status   | file                                    |
|-----------|----------|-----------------------------------------|
| wasp      | CHANGE   | conda/wasp/meta.yaml                    |
| wasp      | CHANGE   | conda/wasp/pyhit.patch                  |
| wasp      | NEW      | framework/contrib/hit                   |
| wasp      | CHANGE   | framework/contrib/wasp                  |
| wasp      | CHANGE   | python/pyhit/pyhit.py                   |
| moose-dev | CHANGE   | conda/moose-dev/conda_build_config.yaml |
| moose-dev | CHANGE   | conda/moose-dev/meta.yaml               |
| moose-dev | CHANGE   | framework/contrib/neml2                 |

### Versioner versions

Found 9 packages, 2 changed, 0 failed

| package   | status   | hash    | to hash   | version            | to version         |
|-----------|----------|---------|-----------|--------------------|--------------------|
| wasp      | CHANGE   | 3e37a23 | 33ce26b   | 2025.05.13 build 1 | 2025.06.20 build 0 |
| moose-dev | CHANGE   | 6416dfc | 4f34b54   | 2025.06.13         | 2025.06.20         |

loganharbour committed:
Remove unused capture

refs #30778

loganharbour committed:
Correct issues with rebase

refs #30778

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3028103/), step Versioner verify on a520ff6 wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 58 influential files, 7 changed, 0 added, 0 removed

| package   | status   | file                                    |
|-----------|----------|-----------------------------------------|
| wasp      | CHANGE   | conda/wasp/meta.yaml                    |
| wasp      | CHANGE   | conda/wasp/pyhit.patch                  |
| wasp      | CHANGE   | framework/contrib/wasp                  |
| wasp      | CHANGE   | python/pyhit/pyhit.py                   |
| moose-dev | CHANGE   | conda/moose-dev/conda_build_config.yaml |
| moose-dev | CHANGE   | conda/moose-dev/meta.yaml               |
| moose-dev | CHANGE   | framework/contrib/neml2                 |

### Versioner versions

Found 9 packages, 2 changed, 0 failed

| package   | status   | hash    | to hash   | version            | to version         |
|-----------|----------|---------|-----------|--------------------|--------------------|
| wasp      | CHANGE   | 3e37a23 | b528290   | 2025.05.13 build 1 | 2025.07.08 build 1 |
| moose-dev | CHANGE   | b5c6af5 | 3ed4767   | 2025.06.26         | 2025.07.08         |

loganharbour committed:
Fix ambiguous call

refs #30778

loganharbour committed:
Re-add call to registerAll

refs #30778

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3028357/), step Versioner verify on 62cb75e wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 58 influential files, 7 changed, 0 added, 0 removed

| package   | status   | file                                    |
|-----------|----------|-----------------------------------------|
| wasp      | CHANGE   | conda/wasp/meta.yaml                    |
| wasp      | CHANGE   | conda/wasp/pyhit.patch                  |
| wasp      | CHANGE   | framework/contrib/wasp                  |
| wasp      | CHANGE   | python/pyhit/pyhit.py                   |
| moose-dev | CHANGE   | conda/moose-dev/conda_build_config.yaml |
| moose-dev | CHANGE   | conda/moose-dev/meta.yaml               |
| moose-dev | CHANGE   | framework/contrib/neml2                 |

### Versioner versions

Found 9 packages, 2 changed, 0 failed

| package   | status   | hash    | to hash   | version            | to version         |
|-----------|----------|---------|-----------|--------------------|--------------------|
| wasp      | CHANGE   | 3e37a23 | b528290   | 2025.05.13 build 1 | 2025.07.08 build 1 |
| moose-dev | CHANGE   | b5c6af5 | 3ed4767   | 2025.06.26         | 2025.07.08         |

loganharbour committed:
More fixes after rebase

refs #30778

loganharbour committed:
Remove single quotes so that there is less to patch in apps

refs #30778

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3029200/), step Versioner verify on af73494 wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 58 influential files, 7 changed, 0 added, 0 removed

| package   | status   | file                                    |
|-----------|----------|-----------------------------------------|
| wasp      | CHANGE   | conda/wasp/meta.yaml                    |
| wasp      | CHANGE   | conda/wasp/pyhit.patch                  |
| wasp      | CHANGE   | framework/contrib/wasp                  |
| wasp      | CHANGE   | python/pyhit/pyhit.py                   |
| moose-dev | CHANGE   | conda/moose-dev/conda_build_config.yaml |
| moose-dev | CHANGE   | conda/moose-dev/meta.yaml               |
| moose-dev | CHANGE   | framework/contrib/neml2                 |

### Versioner versions

Found 9 packages, 2 changed, 0 failed

| package   | status   | hash    | to hash   | version            | to version         |
|-----------|----------|---------|-----------|--------------------|--------------------|
| wasp      | CHANGE   | 3e37a23 | b528290   | 2025.05.13 build 1 | 2025.07.08 build 1 |
| moose-dev | CHANGE   | b5c6af5 | 3ed4767   | 2025.06.26         | 2025.07.08         |

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3042445/), step Versioner verify on 3806a7f wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 58 influential files, 7 changed, 0 added, 0 removed

| package   | status   | file                                    |
|-----------|----------|-----------------------------------------|
| wasp      | CHANGE   | conda/wasp/meta.yaml                    |
| wasp      | CHANGE   | conda/wasp/pyhit.patch                  |
| wasp      | CHANGE   | framework/contrib/wasp                  |
| wasp      | CHANGE   | python/pyhit/pyhit.py                   |
| moose-dev | CHANGE   | conda/moose-dev/conda_build_config.yaml |
| moose-dev | CHANGE   | conda/moose-dev/meta.yaml               |
| moose-dev | CHANGE   | framework/contrib/neml2                 |

### Versioner versions

Found 9 packages, 2 changed, 0 failed

| package   | status   | hash    | to hash   | version            | to version         |
|-----------|----------|---------|-----------|--------------------|--------------------|
| wasp      | CHANGE   | 3e37a23 | b528290   | 2025.05.13 build 1 | 2025.07.08 build 1 |
| moose-dev | CHANGE   | b5c6af5 | 3ed4767   | 2025.06.26         | 2025.07.08         |

dschwen commented on framework/src/base/MooseServer.C:
```diff
@@ -1517,30 +1611,99 @@ MooseServer::getRequiredParamsText(const std::string & subblock_path,
 
-bool
-MooseServer::rootIsValid() const
+// TODO RUN PARSER PARSE FIRST !!
```
should there be an assertion to make sure the parser was run?

dschwen commented on framework/src/base/MooseServer.C:
```diff
@@ -1517,30 +1611,99 @@ MooseServer::getRequiredParamsText(const std::string & subblock_path,
 }
 
 const std::string &
-MooseServer::getDocumentText() const
```
why not const anymore?

dschwen commented on framework/src/base/ResidualObject.C:
```diff
@@ -38,13 +38,12 @@ ResidualObject::ResidualObject(const InputParameters & parameters, bool is_nodal
                     parameters.get<THREAD_ID>("_tid"),
                     is_nodal),
-    Restartable(this, parameters.get<std::string>("_moose_base") + "s"),
+    Restartable(this, getBase() + "s"),
```
The `+ "s"` seems gratuitous. Not all plural nouns are formed by appending an _s_. This might not be an issue now, but apps might define systems that will result in wrong plural forms. Albeit, just a minor cosmetic issue.

loganharbour replied:
But... this wasn't me

dschwen reviewed (commented):
Minor comments

loganharbour commented on framework/include/parser/CommandLine.h:
```diff
@@ -233,6 +233,8 @@ class CommandLine
   /// Whether or not command line parameters have been populated
   bool _command_line_params_populated = false;
+  /// Whether or not the HIT parameters have been built (set as used)
+  bool _hit_params_built = false;
```
I'm just copying my friends above

loganharbour committed:
Clean up const correctness in MooseServer

loganharbour committed:
Correct docstrings

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3058839/), step Versioner verify on fdb3255 wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 58 influential files, 7 changed, 0 added, 0 removed

| package   | status   | file                                    |
|-----------|----------|-----------------------------------------|
| wasp      | CHANGE   | conda/wasp/meta.yaml                    |
| wasp      | CHANGE   | conda/wasp/pyhit.patch                  |
| wasp      | CHANGE   | framework/contrib/wasp                  |
| wasp      | CHANGE   | python/pyhit/pyhit.py                   |
| moose-dev | CHANGE   | conda/moose-dev/conda_build_config.yaml |
| moose-dev | CHANGE   | conda/moose-dev/meta.yaml               |
| moose-dev | CHANGE   | framework/contrib/neml2                 |

### Versioner versions

Found 9 packages, 2 changed, 0 failed

| package   | status   | hash    | to hash   | version            | to version         |
|-----------|----------|---------|-----------|--------------------|--------------------|
| wasp      | CHANGE   | 3e37a23 | b528290   | 2025.05.13 build 1 | 2025.07.08 build 1 |
| moose-dev | CHANGE   | b5c6af5 | 3ed4767   | 2025.06.26         | 2025.07.08         |

loganharbour committed:
Correct min gcc compiler error

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3061653/), step Versioner verify on 68e2aa1 wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 58 influential files, 7 changed, 0 added, 0 removed

| package   | status   | file                                    |
|-----------|----------|-----------------------------------------|
| wasp      | CHANGE   | conda/wasp/meta.yaml                    |
| wasp      | CHANGE   | conda/wasp/pyhit.patch                  |
| wasp      | CHANGE   | framework/contrib/wasp                  |
| wasp      | CHANGE   | python/pyhit/pyhit.py                   |
| moose-dev | CHANGE   | conda/moose-dev/conda_build_config.yaml |
| moose-dev | CHANGE   | conda/moose-dev/meta.yaml               |
| moose-dev | CHANGE   | framework/contrib/neml2                 |

### Versioner versions

Found 9 packages, 2 changed, 0 failed

| package   | status   | hash    | to hash   | version            | to version         |
|-----------|----------|---------|-----------|--------------------|--------------------|
| wasp      | CHANGE   | 3e37a23 | b528290   | 2025.05.13 build 1 | 2025.07.08 build 1 |
| moose-dev | CHANGE   | b5c6af5 | 3ed4767   | 2025.06.26         | 2025.07.08         |

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/3061702/), step Verify coverage on 68e2aa1 wanted to post the following:


The following coverage requirement(s) failed:
- Failed to generate `chemical_reactions` coverage rate (required: 92.0%)
- Failed to generate `contact` coverage rate (required: 87.0%)
- Failed to generate `electromagnetics` coverage rate (required: 94.0%)
- Failed to generate `external_petsc_solver` coverage rate (required: 84.0%)
- Failed to generate `fluid_properties` coverage rate (required: 84.0%)
- Failed to generate `fsi` coverage rate (required: 85.0%)
- Failed to generate `functional_expansion_tools` coverage rate (required: 81.0%)
- Failed to generate `geochemistry` coverage rate (required: 96.0%)
- Failed to generate `heat_transfer` coverage rate (required: 87.0%)
- Failed to generate `level_set` coverage rate (required: 85.0%)
- Failed to generate `misc` coverage rate (required: 32.0%)
- Failed to generate `navier_stokes` coverage rate (required: 77.0%)
- Failed to generate `optimization` coverage rate (required: 86.0%)
- Failed to generate `peridynamics` coverage rate (required: 77.0%)
- Failed to generate `phase_field` coverage rate (required: 85.0%)
- Failed to generate `porous_flow` coverage rate (required: 95.0%)
- Failed to generate `ray_tracing` coverage rate (required: 93.0%)
- Failed to generate `rdg` coverage rate (required: 63.0%)
- Failed to generate `reactor` coverage rate (required: 90.0%)
- Failed to generate `scalar_transport` coverage rate (required: 81.0%)
- Failed to generate `solid_mechanics` coverage rate (required: 84.0%)
- Failed to generate `solid_properties` coverage rate (required: 83.0%)
- Failed to generate `stochastic_tools` coverage rate (required: 88.0%)
- Failed to generate `subchannel` coverage rate (required: 87.0%)
- Failed to generate `thermal_hydraulics` coverage rate (required: 88.0%)
- Failed to generate `xfem` coverage rate (required: 80.0%)

dschwen reviewed (approved):
Looks fine to me!

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3092567/), step Versioner verify on 21787e9 wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 58 influential files, 7 changed, 0 added, 0 removed

| package   | status   | file                                    |
|-----------|----------|-----------------------------------------|
| wasp      | CHANGE   | conda/wasp/meta.yaml                    |
| wasp      | CHANGE   | conda/wasp/pyhit.patch                  |
| wasp      | CHANGE   | framework/contrib/wasp                  |
| wasp      | CHANGE   | python/pyhit/pyhit.py                   |
| moose-dev | CHANGE   | conda/moose-dev/conda_build_config.yaml |
| moose-dev | CHANGE   | conda/moose-dev/meta.yaml               |
| moose-dev | CHANGE   | framework/contrib/neml2                 |

### Versioner versions

Found 9 packages, 2 changed, 0 failed

| package   | status   | hash    | to hash   | version            | to version         |
|-----------|----------|---------|-----------|--------------------|--------------------|
| wasp      | CHANGE   | 3e37a23 | ec9603a   | 2025.05.13 build 1 | 2025.07.08 build 0 |
| moose-dev | CHANGE   | 46da177 | 323ed7c   | 2025.07.22         | 2025.07.28         |

loganharbour committed:
Require that all objects have a type and name

loganharbour committed:
Improve parsed material error context

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3096554/), step Versioner verify on 2f56f68 wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 58 influential files, 7 changed, 0 added, 0 removed

| package   | status   | file                                    |
|-----------|----------|-----------------------------------------|
| wasp      | CHANGE   | conda/wasp/meta.yaml                    |
| wasp      | CHANGE   | conda/wasp/pyhit.patch                  |
| wasp      | CHANGE   | framework/contrib/wasp                  |
| wasp      | CHANGE   | python/pyhit/pyhit.py                   |
| moose-dev | CHANGE   | conda/moose-dev/conda_build_config.yaml |
| moose-dev | CHANGE   | conda/moose-dev/meta.yaml               |
| moose-dev | CHANGE   | framework/contrib/neml2                 |

### Versioner versions

Found 9 packages, 2 changed, 0 failed

| package   | status   | hash    | to hash   | version            | to version         |
|-----------|----------|---------|-----------|--------------------|--------------------|
| wasp      | CHANGE   | 3e37a23 | ec9603a   | 2025.05.13 build 1 | 2025.07.08 build 0 |
| moose-dev | CHANGE   | 46da177 | 323ed7c   | 2025.07.22         | 2025.07.28         |

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3098930/), step Versioner verify on f45a8c4 wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 58 influential files, 7 changed, 0 added, 0 removed

| package   | status   | file                                    |
|-----------|----------|-----------------------------------------|
| wasp      | CHANGE   | conda/wasp/meta.yaml                    |
| wasp      | CHANGE   | conda/wasp/pyhit.patch                  |
| wasp      | CHANGE   | framework/contrib/wasp                  |
| wasp      | CHANGE   | python/pyhit/pyhit.py                   |
| moose-dev | CHANGE   | conda/moose-dev/conda_build_config.yaml |
| moose-dev | CHANGE   | conda/moose-dev/meta.yaml               |
| moose-dev | CHANGE   | framework/contrib/neml2                 |

### Versioner versions

Found 9 packages, 2 changed, 0 failed

| package   | status   | hash    | to hash   | version            | to version         |
|-----------|----------|---------|-----------|--------------------|--------------------|
| wasp      | CHANGE   | 3e37a23 | ec9603a   | 2025.05.13 build 1 | 2025.07.08 build 0 |
| moose-dev | CHANGE   | 46da177 | 323ed7c   | 2025.07.22         | 2025.07.28         |

loganharbour committed:
Add AppFactory::create method to be used in unit tests

loganharbour committed:
Use new helper for unit test apps

loganharbour committed:
Re-add createShared so that griffin builds

loganharbour committed:
Bring back unique action name; Griffin needs it

loganharbour committed:
Use mooseError in static namespace

refs #30778

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3121730/), step Versioner verify on 81cf51e wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 58 influential files, 7 changed, 0 added, 0 removed

| package   | status   | file                                    |
|-----------|----------|-----------------------------------------|
| wasp      | CHANGE   | conda/wasp/meta.yaml                    |
| wasp      | CHANGE   | conda/wasp/pyhit.patch                  |
| wasp      | CHANGE   | framework/contrib/wasp                  |
| wasp      | CHANGE   | python/pyhit/pyhit.py                   |
| moose-dev | CHANGE   | conda/moose-dev/conda_build_config.yaml |
| moose-dev | CHANGE   | conda/moose-dev/meta.yaml               |
| moose-dev | CHANGE   | framework/contrib/neml2                 |

### Versioner versions

Found 9 packages, 2 changed, 0 failed

| package   | status   | hash    | to hash   | version            | to version         |
|-----------|----------|---------|-----------|--------------------|--------------------|
| wasp      | CHANGE   | 3e37a23 | ec9603a   | 2025.05.13 build 1 | 2025.07.08 build 0 |
| moose-dev | CHANGE   | 46da177 | 323ed7c   | 2025.07.22         | 2025.07.28         |

loganharbour committed:
Fix python versions for pyhit

loganharbour committed:
Change builder call after rebase

refs #30778

hugary1995 committed:
Update update_and_rebuild_neml2.sh

- Perform a neml2 submodule update
- Change a few cmake definitions in the update_and_rebuild_neml2.sh and configure_neml2.sh scripts
- Update configure.ac and autoreconf

ref #29579

loganharbour committed:
Update neml2 again to get wasp and hit bump

refs #30778

loganharbour committed:
Bump moose-dev and moose-wasp

moose-wasp bump is due to changes in hit and pyhit

moose-dev bump is due to updating neml2

refs #30778 #29579

loganharbour committed:
Update versioner_hashes

loganharbour commented:
Failures are related and patches are ready (see starting comment).
